File: c/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    // Prompt the user for some text
    string text = get_string("Enter your text please:\n");
    int text_len = strlen(text);

    // Count the number of letters, words, and sentences in the text
    char punc[] = {'.', '!', '?'};
    int PUNC_LEN = 3;
    int letter_c = 0;
    int word_c = 1;
    int sentence_c = 0;
    for (int i = 0; i < text_len; i++)
    {
        if (isalpha(text[i]))
            letter_c++;
        else if (text[i] == ' ')
            word_c++;
        else
            for (int j = 0; j < PUNC_LEN; j++)
            {
                if (text[i] == punc[j])
                {
                    sentence_c++;
                    break;
                }
            }
    }
    printf("letters: %i\nwords: %i\nsentences: %i\n", letter_c, word_c, sentence_c);

    // Compute the Coleman-Liau index
    float L = (float) letter_c / word_c * 100;
    float S = (float) sentence_c / word_c * 100;
    int index = round(0.0588 * L - 0.296 * S - 15.8);

    // Print the grade level
    if (index < 1)
        printf("Before Grade 1\n");
    else if (index > 16)
        printf("Grade 16+\n");
    else
        printf("Grade %i\n", index);
}

File: python/sentimental-readability/readability.py
# count the letters, words and setences and return them
def counter(text):
    punc = [".", "?", "!"]
    letter_c = 0
    word_c = 0
    sentence_c = 0
    for c in range(len(text)):
        if text[c].isalpha():
            letter_c += 1
        elif text[c] in punc:
            sentence_c += 1
            word_c += 1
        elif text[c] == " " and text[c - 1] not in punc:
            word_c += 1
    return letter_c, word_c, sentence_c


# calculate the index
def coleman_liau(l, w, s):
    L = l / w * 100
    S = s / w * 100
    index = round(0.0588 * L - 0.296 * S - 15.8)
    return index


def main():
    text = input("Text: ")
    l, w, s = counter(text)
    i = coleman_liau(l, w, s)
    # based on the index, decide the readability level
    if i < 1:
        print("Before Grade 1")
    elif i > 16:
        print("Grade 16+")
    else:
        print(f"Grade {i}")


main()
